queries/scala/highlights.scm

```diff
@@ -0,0 +1,52 @@
+;; keywords
+
+[
+  "abstract"
+  "case"
+  "catch"
+  "class"
+  "def"
+  "do"
+  "extends"
+  "final"
+  "finally"
+  "forSome"
+  "implicit"
+  "lazy"
+  "macro"
+  "match"
+  "new"
+  "null"
+  "object"
+  "override"
+  "package"
+  "private"
+  "protected"
+  "sealed"
+  "super"
+  "this"
+  "throw"
+  "trait"
+  "try"
+  "type"
+  "val"
+  "var"
+  "with"
+  "yield"
+] @keyword
+
+;; special keywords
+
+[
+  "else"
+  "if"
+] @conditional
+
+[
+  "for" 
+  "while"
+] @repeat
+
+"import" @includes
+
+"return" @keyword.return
```

Comment:
You can also add `yield` to that category.

Reply:
well `yield` is really not a return statement

Reply:
ya it's just that in other languages (e.g. python or javascript) yield is use to indicate that the current function is _generator function_.

in scala, you can do
```scala
def myFunction(): List[Int] = {
  val nums = List(1,2,3)
  val moreNums = for (n <- nums) yield n + 1
  1 :: moreNums // returns List(1,2,3,4)
}
```

`yield` is more of a special word related to `for`.